### PR TITLE
Remove power distribution results from `Report` objects

### PR DESCRIPTION
**Pull request by shsms**
Instead, this PR makes the power distribution results available directly from the `*Pool`s.

This should go some way towards reducing the frequency of `Report` objects as reported in #818

### CODE REVIEW

**Comment by shsms** [comment]
Draft until channel release with `filter` on receivers.

**Comment by shsms** [comment]
Based on #1000.

**Comment by shsms** [comment]
This is now ready.  `ReceiverFetcherWith` is internal, but still could use a better name I think, if anyone has any ideas.

**Comment by llucax** [comment]
> This is now ready. `ReceiverFetcherWith` is internal, but still could use a better name I think, if anyone has any ideas.

Who doesn't like to get involved in a nice and long name discussion? :laughing:

File: src/frequenz/sdk/_internal/_channels.py [hunk 1]
```diff
@@ -29,6 +31,37 @@ def new_receiver(self, *, limit: int = 50) -> Receiver[T_co]:
         """
 
 
+class ReceiverFetcherWith(ReceiverFetcher[U_co], typing.Generic[T_co, U_co]):
```
Comment (llucax) [inline src/frequenz/sdk/_internal/_channels.py]: So about the name discussion, I would mimic `Receiver` and say this is a `MappingReceiverFetcher`, and call the `manipulator` a `mapping_function`?

Comment (llucax) [inline src/frequenz/sdk/_internal/_channels.py]:
```suggestion
class ReceiverFetcherWith(typing.Generic[T_co, U_co]):
```

Inheriting from a protocol will make this a protocol, but we actually don't want this to be a protocol in itself, just a regular class that happens to implement the `ReceiverFetcher` protocol, right?

File: pyproject.toml [hunk 2]
```diff
@@ -30,7 +30,7 @@ dependencies = [
   # changing the version
   # (plugins.mkdocstrings.handlers.python.import)
   "frequenz-client-microgrid >= 0.4.0, < 0.5.0",
-  "frequenz-channels >= 1.0.0-rc1, < 2.0.0",
+  "frequenz-channels >= 1.1.0, < 2.0.0",
```
Comment (llucax) [inline pyproject.toml:33]: You must be happy about finally getting rid of the `-rc1` :laughing:

Just as a heads up, once we reach *the real 1.0.0* :tm: for the SDK, we can't bump minimum dependencies anymore.

Also, while you are upgrading, wouldn't it make sense to also replace the `LatestValueCache` with the one in channels? But OK to do in a separate PR too.

**Review by llucax — COMMENTED** [review]
Just a few small comments.

File: src/frequenz/sdk/_internal/_channels.py (hunk 1 above)
Comment (shsms) [reply src/frequenz/sdk/_internal/_channels.py]: changed.

Comment (shsms) [reply src/frequenz/sdk/_internal/_channels.py]: true, it should be fine without deriving from it.  changed.

File: pyproject.toml (hunk 2 above)
Comment (shsms) [reply pyproject.toml:33]: > You must be happy about finally getting rid of the -rc1 😆

properly happy. :-)

> Also, while you are upgrading, wouldn't it make sense to also replace the LatestValueCache with the one in channels? But OK to do in a separate PR too.

will do it separately.

**Review by llucax — APPROVED** [review]
